ReSwift/CoreTypes/Store.swift

```diff
@@ -28,7 +28,7 @@ open class Store<State: StateType>: StoreType {
             subscriptions.forEach {
                 // if a selector is available, subselect the relevant state
                 // otherwise pass the entire state to the subscriber
-                $0.subscriber?._newState(state: $0.selector?(state) ?? state)
+                $0.subscriber?._newState(state: $0.selector?(state) ?? state!)
```

Comment:
I'm only new to the codebase, so please excuse me if you already have defined standards around this, but it might be worthwhile using a `guard` with a `preconditionFailure()` here instead of the forced unwrap (`!`), so that you can provide a meaningful error message if the user of this API ever hits a condition where `state` is nil.

Something like:

```swift
guard let state = $0.selector?(state) ?? state else {
    preconditionFailure("Unexpectedly found `nil` while unwrapping state") // This could obviously be more descriptive
}

$0.subscriber?._newState(state: state)
```